common/membership/static/fx.go

```diff
@@ -0,0 +1,46 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package static
+
+import (
+	"go.temporal.io/server/common/primitives"
+	"go.uber.org/fx"
+
+	"go.temporal.io/server/common/membership"
+)
+
+func MembershipModule(
+	hosts map[primitives.ServiceName][]string,
+) fx.Option {
+	return fx.Options(
+		fx.Provide(func() membership.Monitor {
+			return newStaticMonitor(hosts)
+		}),
+		fx.Provide(func(serviceName primitives.ServiceName) membership.HostInfoProvider {
+			hostInfo := membership.NewHostInfoFromAddress(hosts[serviceName][0])
```

Comment:
This means this only supports one member per service? Could we support more than one by accepting the self-address also?

Or if not, at least panic or error if we find more than one here.

Reply:
> This means this only supports one member per service?

Right; I moved that code here from the tests. onebox allows multiple hosts, but only ever uses the first one (there is a comment mentioning that should be fixed, but that seems to require more effort).

> Could we support more than one by accepting the self-address also?

I'm not sure I follow. Could you describe what role that address would play / how it would be used here?

> Or if not, at least panic or error if we find more than one here.

I might have to change the onebox setup for this as it is passing in multiple History hosts.

Reply:
HostInfoProvider is answering the question "who am I?" Apparently this code was wrong in onebox (I didn't realize): all three history nodes were saying they were the first one. It probably doesn't matter for most things but I know at least worker would break in some ways.

So I think the caller of this static stuff needs to provide the total service->address list mapping, plus one extra address which is its own address (which should be in the list of addresses for that service type, and that should probably be validated)